.github/workflows/test.yml

```diff
@@ -1,23 +1,24 @@
 name: test suite
 
-on:
-  push:
-    branches: [main]
-  pull_request:
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
 
 jobs:
   test:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", pypy-3.9]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", pypy-3.9]
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
+        allow-prereleases: true
```

Comment:
Why would I want this?

Reply:
Python 3.12 is a pre-release, the full release is in October. This allows us to have `3.12` in `matrix.python-version` instead of `3.12-dev`.

https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases

It doesn't matter which one is used. Some people don't like having to edit out the `-dev` when the full release is made. Which do you prefer?

Reply:
Won't this apply to other Python versions too? Wouldn't it essentially test against 3.11-dev too with this turned on?

Reply:
Yes, although GitHub Actions sticks to official releases. They all have newer _full release_ patch versions out, so will never go back to an older pre-release patch.

> Use `-dev` instead of a patch number (e.g., `3.12-dev`) to install the latest patch version release for a given minor version, _alpha and beta releases included_.

https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#available-versions-of-python-and-pypy

So `3.11-dev` currently installs 3.11.3.

GitHub Actions is different to Travis CI in this regard, where `-dev` pointed to a nightly build (so `3.11-dev` would be a recent commit on the `3.11` branch).

Reply:
Ok, I just read the documentation for this option. What I cannot understand is why it's not the default.